### ISSUE TITLE
Minor inconsistency in error message

### ISSUE THREAD

**Issue by hamogu** [sherpa/sherpa#244]
Today  I saw this:

```
sherpa-57> save_arrays("stuff.tbl", [energies[:-1], energies[1:], flux[:-1]], ["keV","photons/s/cm**2/keV"], ascii=True)
IOErr: Received more column names: 2, than required: 3
```

Surely, the error message should should say that it received "FEWER" column names, not "MORE" since 2 < 3 for most definitions of numbers...

**Comment by DougBurke** [comment]
This happens for both crates and astropy backends, since they both have lines like the following (see `set_arrays` in sherpa/astro/io/<xxx>_backend.py, although the error doesn't necessarily occur under the same conditions)L

```
    if len(args) != len(fields):
        raise IOErr('toomanycols', str(len(fields)), str(len(args)))
```

It would probably be easiest to rename this error and rewrite so that it's just a "number of column names does not match" rather than have separate less-than or more-than errors.

**Comment by hamogu** [comment]
Agreed. No need to say less or more. People will figure that out.

**Comment by DougBurke** [comment]
With the master branch, the `toomanycols` error is only used in a few places, so it should be an easy change, once python 3 (PR #229) has landed.

```
% find sherpa -name \*py -exec grep toomanycols /dev/null {} \;
sherpa/utils/err.py:            'toomanycols': "Received more column names: %s, than required: %s",
sherpa/io.py:        raise IOErr('toomanycols')
sherpa/astro/io/crates_backend.py:        raise IOErr('toomanycols', str(len(fields)), str(len(args)))
sherpa/astro/io/pyfits_backend.py:        raise IOErr("toomanycols", str(len(fields)), str(len(args)))
sherpa/ui/utils.py:            raise sherpa.utils.err.IOErr('toomanycols')
sherpa/ui/utils.py:                raise sherpa.utils.err.IOErr('toomanycols', str(tnames), '2')
```

Changing the message and error "type" is potentially a breaking change, but it's one I can argue for.

**Comment by dtnguyen2** [comment]
Trying to reproduce the original issue:

```
sherpa-4.14-dev) [dtn@devel12 sherpa]$ cat tmp.py
import numpy as np
import pdb
from sherpa.astro.ui import *
tmp=np.arange(10)
#pdb.set_trace()
save_arrays("tmp.txt",[[tmp, tmp, tmp]], ["a", "b"], clobber=True)
```

Let's make sure there is no file ```tmp.txt```
```
(sherpa-4.14-dev) [dtn@devel12 sherpa]$ rm -f tmp.txt
```

Shouldn't the following throw an exception:
```
(sherpa-4.14-dev) [dtn@devel12 sherpa]$ python tmp.py
/export/miniconda/envs/sherpa-4.14-dev/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: Unable to load the ciao_version module to determine version number- defaulting 'group' version to 0.0.0
  return f(*args, **kwds)
```
instead of producing the file:

```
(sherpa-4.14-dev) [dtn@devel12 sherpa]$ cat tmp.txt 
#a b
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
```

Reduce the number of columns to be < the fields;

```
sherpa-4.14-dev) [dtn@devel12 sherpa]$ cat tmp.py
import numpy as np
import pdb
from sherpa.astro.ui import *
tmp=np.arange(10)
#pdb.set_trace()
save_arrays("tmp.txt",[[tmp]], ["a", "b"], clobber=True)
```

Also works:
```
(sherpa-4.14-dev) [dtn@devel12 sherpa]$ cat tmp.txt 
#a b
0 1 2 3 4 5 6 7 8 9
```

What am I missing?  Kind of odd in light of PR #1136

**Comment by DougBurke** [comment]
@dtnguyen2 - can you open this as a new issue because it'll get lost here.

**Comment by dtnguyen2** [comment]
Closed, see PR #1136.  As for my the inconsistent behavior of ```save_arrays``` when ascii=True/False, see issue #1251

**Comment by DougBurke** [comment]
Ta
